moler/cmd/at/get_cell_id_nr.py

```diff
@@ -0,0 +1,277 @@
+        Verify with latest 3gpp specification 27.007. Example output:
+
+        +C5GREG: <n>,<stat>[,[<lac>],[<ci>],[<AcT>],[<Allowed_NSSAI_length>], [<Allowed_NSSAI>][,[<cause_type>],
+        [<reject_cause>]][,[<cag_stat>][,<caginfo>]
+
+        OK
+
+        where <n> Possible options for selected_mode, but verify with latest 3gpp specification 27.007.
+
+        :param connection: Moler connection to device, terminal when command is executed.
+        :param prompt: prompt where we start from.
+        :param newline_chars: Characters to split local lines - list.
+        :param runner: Runner to run command.
+        """
+        super(GetCellIdNr, self).__init__(connection, operation='execute', prompt=prompt,
+                                          newline_chars=newline_chars, runner=runner)
+
+    def build_command_string(self):
+        return "AT+C5GREG?"
+
+    def on_new_line(self, line, is_full_line):
+        """
+        Method to parse command output. Will be called after line with command echo.
+
+        +C5GREG: <n>,<stat>[,[<tac>],[<ci>],[<AcT>],[<Allowed_NSSAI_length>],[<Allowed_NSSAI>]...
+                 ...[,<cause_type>,<reject_cause>]][,<cag_stat>][,<caginfo>]
+
+        OK
+
+        Write your own implementation but don't forget to call on_new_line from base class
+
+        :param line: Line to parse, new lines are trimmed
+        :param is_full_line:  False for chunk of line; True on full line (NOTE: new line character removed)
+        :return: None
+        """
+        if is_full_line:
+            try:
+                self._parse_cell_data_nr(line)
+            except ParsingDone:
+                pass
+        return super(GetCellIdNr, self).on_new_line(line, is_full_line)
+
+    # <n>=1, where +C5GREG: <n>[,[<tac>],[<ci>],[<AcT>],[<Allowed_NSSAI_length>],[<Allowed_NSSAI>]]
+    # +C5GREG: 1,"0F6B0578",7,2,"aB" - option regarding to manual tests (different than 3gpp)
+    _re_cell_id_mode_1 = \
+        re.compile(r'^\s*\+C5GREG:\s(?P<n>([0-9]+)),\"(?P<tac>([0-9A-Fa-f]*))\"(,)?(\")?(?P<ci>([0-9A-Fa-f]*))?(\")?'
+                   r'(,)?(?P<AcT>([0-9]{1,2}))?(,)?(?P<Allowed_NSSAI_length>([0-9]+))?(,)?'
+                   r'(\")?(?P<Allowed_NSSAI>([0-9A-Za-z.]*))?(\")?.*$')
+
+    # +C5GREG: <n>
+    _re_cell_id_mode = re.compile(r'^\s*\+C5GREG:\s(?P<n>([0-9]+)).*$')
+
+    # +C5GREG: <raw_output>
+    _re_raw_data = re.compile(r'^\s*\++C5GREG:\s(?P<raw_output>.*)')
+
+    def _parse_cell_data_nr(self, line):
+        """
+        Parse network registration status and location information:
+        +C5GREG: 1,11
+        +C5GREG: 1,"54DB","0F6B0578",7,2,"aB" - extra
+        +C5GREG: 2,5,"54DB","0F6B0578",7,2,"aB"
+        +C5GREG: 3,5,,,,,,1,2
+        +C5GREG: 3,5,"54DB","0F6B0578",7,2,"aB",1,2
+        +C5GREG: 4,5,,,,,,1,2
+        +C5GREG: 5,5,"54DB","0F6B0578",7,2,"aB",1,2,123,"info"
+        """
+        results_keys = {
```

Comment:
here seems that order is the same for all variants. <br>
Maybe do similar like in `get_cell_id`
` results_keys = ['n', 'stat', 'tac', 'ci', 'AcT', 'Allowed_NSSAI_length', 'Allowed_NSSAI', 'cause_type', 'reject_cause',
                'cag_stat', 'caginfo']`

Reply:
Yes, You are right. Applied. Thanks!